applications/ecallmgr/src/ecallmgr_call_control.erl

```diff
@@ -821,23 +869,51 @@ handle_dialplan(JObj, #state{call_id=CallId
     of
         'true' ->
             {{'value', Cmd}, NewCmdQ1} = queue:out(NewCmdQ),
-            AppName = kz_json:get_value(<<"Application-Name">>, Cmd),
-            _ = case CallUp
-                    orelse is_post_hangup_command(AppName)
-                of
-                    'true' -> execute_control_request(Cmd, State);
-                    'false' ->
-                        lager:debug("command '~s' is not valid after hangup, ignoring", [AppName]),
-                        maybe_send_error_resp(CallId, Cmd),
-                        self() ! {'force_queue_advance', CallId}
-                end,
-            MsgId = kz_json:get_value(<<"Msg-ID">>, Cmd),
-            State#state{command_q=NewCmdQ1
-                       ,current_app=AppName
-                       ,current_cmd=Cmd
-                       ,keep_alive_ref=get_keep_alive_ref(State)
-                       ,msg_id=MsgId
-                       };
+            AppName = kz_json:get_ne_binary_value(<<"Application-Name">>, Cmd),
```

Comment:
Another section that looks like the two chunks above and could be refactored into a shared function?